ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionValidator.java

```diff
@@ -184,6 +189,14 @@ public ValidationResult<TransactionInvalidReason> validate(
               intrinsicGasCost, transaction.getGasLimit()));
     }
 
+    if (transaction.getTo().isEmpty() && transaction.getPayload().size() > maxInitcodeSize) {
```

Comment:
```suggestion
    if (transaction.isContractCreation() && transaction.getPayload().size() > maxInitcodeSize) {
```

Reply:
Also, think this is missing a test as well?

Reply:
yep.  Added.